### ISSUE TITLE
[508-defect-2] CST: Evidence headers should be unique

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#41818]
## Description

List of additional evidence should be unique or uniquely described.

Related: https://github.com/department-of-veterans-affairs/va.gov-team/issues/41447 (Under "Should" fix)

![List of additional evidence should be unique or uniquely described](https://user-images.githubusercontent.com/14154792/169905053-5d1b9fe2-3873-4463-a074-e3a01da0f4b3.png)

## Tasks
@jacobworrell updated this ticket to represent its current status and path to closure ✨ 

### [Technical discovery](https://github.com/department-of-veterans-affairs/va.gov-team/issues/47246)
- [ ] Complete technical discovery ticket

### Design
- [ ] Update design and content based on technical discovery ticket @skylerschain 

### Development
- [ ] Add unique description for each entry
- [ ] Add tests (as needed)

## Definition of done

- [ ] All tasks complete
- [ ] All tests passing

**Comment by jaredcunha** [comment]
Any chance a table might be better-suited to this? The screenshot above doesn't quite demonstrate if there are additional states of the UI, but with so as much repetition, I'm wondering if a table might improve navigation and scan-ability.

**Comment by Mottie** [comment]
The [`va-table` web component](https://design.va.gov/storybook/?path=/docs/components-va-table--default), isn't currently set up to work with pagination. Also, I don't think we have a design pattern established for using tables for documents. Please run this by platform design before switching to use a table.

**Comment by sortizsh** [comment]
Should go back to Design to look at the upload screen.

**Comment by skylerschain** [comment]
@Mottie is there a design spec you're working off for these? If so, could you point me towards it?

**Comment by Mottie** [comment]
There is no design, this ticket was part of an accessibility review. So there isn't a design spec that I could find for this.

**Comment by skylerschain** [comment]
Why does the first item in the list have the date in it, while the second and third items only say "Additional evidence"?

**Comment by jerekshoe** [comment]
First item is a suggestion I believe. In production, the heading for each item is simply "Additional Evidence".

**Comment by skylerschain** [comment]
Proposed solution:
– Number the headers to differentiate them
– Update the section header to "Additional evidence submitted" since "submit" is the verb we use above, not "file" (previous version).

![UniqueHeaders](https://user-images.githubusercontent.com/95102058/175059138-29cad7d2-2f09-4030-af1f-da2e057e3349.png)

**Comment by skylerschain** [comment]
Yeah I would love to get a content design check on this. Do you know who that would be?

**Comment by joshkimux** [comment]
@mnorthuis sorry to bug you again, would you be willing to give this a final check before we close this? Specifically, Skyler is proposing to use numbers to indicate the order of items in the additional evidence submitted list 🙇 

Some things that come to mind include:
- Would numbering these create confusion between dates or falsely indicate importance? e.g. would the numbers ascend from the oldest or newest claim?
- Are there any possible alternative variations to the heading?

**Comment by katherine-fung** [comment]
Hi @skylerschain @joshkimux @sortizsh @mnorthuis @jerekshoe @Mottie -- 

Thanks for this interesting question! After reviewing this request with the rest of the content team, we recommend wording the headers this way: 

"Additional evidence, 1 of 3" 
"Additional evidence, 2 of 3"  

Can you confirm that it's ok for the unique element to come at the end of the header, instead of the beginning? 

For the section header, we should say "Additional evidence you submitted" (so it's in the active voice, instead of passive).

**Comment by joshkimux** [comment]
@katherine-fung, this is tough. Technically, there is no requirement that the unique bit come at the beginning or end, but from an aural user experience perspective, it can be a bit frustrating to have to listen to "additional evidence" first each time.

Something we may be able to consider (but needs additional verification) would be to use the type (which is required for all submissions) as the unique identifier instead. For example, "STR - Dental, 1 of 2" instead of "Additional Evidence, 1 of 10." If there is only one of a type submitted, it could just be listed as "STR - Dental" without the numerical identifier.

![image](https://user-images.githubusercontent.com/14154792/178516018-60fb2020-c054-44f1-a5f4-125d53a447b2.png)

From a design perspective @skylerschain , I'm also curious if we can look a little bit deeper into:
- Both the median number and extremities of submissions (what's the most common use case and what's the most difficult use case? how might that inform this current bandaid fix vs. future, more meaningful solutions?)

From a QA perspective, I'm a little worried that Veterans can submit a limitless number of evidence with no method of "undoing" or "deleting" a submission. The particular test user I'm looking at (233) has well over 30+ submitted files (which is unlikely, but possible) with no method of removing any of them from their record. @jacobworrell for awareness!

**Comment by katherine-fung** [comment]
Thanks @joshkimux! That's a good point, and I think your proposed idea could work really well. I'm wondering if the label would be even more clear for users if we say, for example, "Medical record - Photocopy, 1 of 2" or "Dental record - Photocopy, 1 of 2" instead of STR. I imagine not everyone will immediately understand what STR means. As long as people easily understand what the document type names mean, using the document type + numerical identifier per your mock-up is great. What do you think?

**Comment by joshkimux** [comment]
@katherine-fung , I would absolutely _love_ if we could do that, but my assumption is that there's some back end or technical constraint related to this. @Mottie is this something we might actually have control of? @jsokohl for awareness 🤞

**Comment by Mottie** [comment]
The "STR - Dental - Photocopy, 1 of 2" is the actual file name that was uploaded. It's readable here, but we have no idea (we'd need to see production data) what the file names actually look like. I've downloaded records from the medical record sitesthat are basically named by the app + date, e.g. `docushare-app-3_11_2020 10_00_00 PM.pdf`, or named by their camera (`IMG_20151113.png`).

**Comment by Mottie** [comment]
Hi @katherine-fung

Yes, I was incorrect. ["STR - Dental - Photocopy" is a document type, and not a file name, that we can change to whatever we want](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/disability-benefits/all-claims/pages/serviceTreatmentRecordsAttachments.js#L53-L54), but the listing in the CST may not accurately be labeled with an `X of Y`, if the same file was uploaded more than once.

**Comment by joshkimux** [comment]
@Mottie and @katherine-fung depending on:
- how much metadata may be added in the future beyond date and filename
- how much additional evidence a Veteran may upload

We could also consider using a plain list (instead of headings) e.g.:
- Service Treatment Record - Dental - Photocopy, uploaded on [date], [filename]
- Service Treatment Record - Dental - Photocopy uploaded on [date], [filename]
- Medical Treatment Record - Non-Government Facility uploaded on [date], [filename]

That would eliminate the issue of duplicative headings and it can still be styled in a way that's easy for sighted folk to parse through. Once again, I think that does depend on how many files we may anticipate a Veteran to upload (what are the edge cases) and the future intention of additional evidence (will they be editable in the future? how will they be structured?)

**Comment by jsokohl** [comment]
- The issue of metadata should be a global consideration, it seems, and not just in context of CST.
- I think we would need to review _all_ the possible document types that we would provide for people to choose and then consider doing smart replacement for acronyms/initialization. That is, the first time the item is uploaded, it's "Service Treatment Record - Dental - Photocopy, uploaded on [date], [filename]" and subsequent ones could be "STR - Dental - Photocopy uploaded on..."

**Comment by katherine-fung** [comment]
Thanks @joshkimux @jsokohl! 

My initial thoughts are that the plain list may present the user with too much information about the documents, without helping them tell each item apart. The headers might be better in that regard. But I will run this by my team and see what they say. 

I agree that we should look at the use of acronyms in this tool and check to see which ones could be confusing for users. Joe's suggestion of using the full phrase on first instance and abbreviating in later instances is already a VA style rule (yay) -- I'll check with my team if it applies to UX copy as well.

**Comment by katherine-fung** [comment]
Hi @joshkimux @Mottie,  

@bethpotts and I chatted about this. We're recommending using [File type, X of Y] as the header, with a few changes.  

1. For the VA form file types, which have longer names (i.e. VA Form 21-0781 - Statement in Support of Claim for PTSD), can we show the name of the file type in the header without the form number? 

i.e. Statement in Support of Claim for PTSD, 1 of 2 

2. We'd like to change instances of the STR acronym in the dropdown and in headers to fully say "Service treatment record," to help users better understand what that means. Can we do that? 

And more generally, we'd love to work with your team to revisit the naming of file types in this tool.

**Comment by jsokohl** [comment]
Perhaps you might eliminate the leading zero. It’s not needed for data or spacing and looks…”technical.”

joe
Joe Sokohl
1.804.873.6964

**Comment by joshkimux** [comment]
@rubensun @skylerschain thoughts on adding this to the design board?

**Comment by skylerschain** [comment]
Yeah, we can add it. Are we aligned on the solution above and we just need to produce mocks? 

@jerekshoe Just want to confirm with you that the above solution looks feasible.

**Comment by skylerschain** [comment]
This should be fixed based on the new [Documents Filed design](https://www.figma.com/file/F8U4wddaFouUPVd4mGBMDI/CST---Evidence-Submission---Spec?type=design&node-id=966-7298&mode=design&t=CAy6JM6Co9OeVB6I-4), which uses file names as headers.
